subsys/bluetooth/audio/shell/bap.c

```diff
@@ -3444,6 +3490,49 @@ static int cmd_create_broadcast_sink(const struct shell *sh, size_t argc, char *
 	return 0;
 }
 
+static int cmd_create_sink_by_name(const struct shell *sh, size_t argc, char *argv[])
+{
+	char *broadcast_name;
+	int err = 0;
+
+	broadcast_name = argv[1];
+	if (!IN_RANGE(strlen(broadcast_name), BT_AUDIO_BROADCAST_NAME_LEN_MIN,
+		      BT_AUDIO_BROADCAST_NAME_LEN_MAX)) {
+
+		shell_error(sh, "Broadcast name should be minimum %d and maximum %d characters",
+			    BT_AUDIO_BROADCAST_NAME_LEN_MIN, BT_AUDIO_BROADCAST_NAME_LEN_MAX);
+
+		return -ENOEXEC;
+	}
+
+	const struct bt_le_scan_param param = {
+		.type = BT_LE_SCAN_TYPE_PASSIVE,
+		.options = BT_LE_SCAN_OPT_NONE,
+		.interval = BT_GAP_SCAN_FAST_INTERVAL,
+		.window = BT_GAP_SCAN_FAST_WINDOW,
+		.timeout = 1000, /* 10ms units -> 10 second timeout */
+	};
+
+	shell_print(sh, "Starting scanning for broadcast_name");
+
+	err = bt_le_scan_start(&param, NULL);
+	if (err) {
+		shell_print(sh, "Fail to start scanning: %d", err);
+
+		return -ENOEXEC;
+	}
+
+	memcpy(auto_scan.broadcast_info.broadcast_name, broadcast_name, strlen(broadcast_name));
+	auto_scan.broadcast_info.broadcast_name[strlen(broadcast_name)] = '\0';
+	utf8_trunc(auto_scan.broadcast_info.broadcast_name);
```

Comment:
```suggestion
	auto_scan.broadcast_info.broadcast_name[strlen(broadcast_name)] = '\0';
```
Since we have a length check earlier we don't need to do a truncate

Reply:
That's true, I'll remove it.